src/algebra/continued_fractions/computation/correctness_terminating.lean

```diff
@@ -0,0 +1,217 @@
+          next_denominator], ac_refl },
+      rw this,
+      -- two calculations needed to show the claim
+      have tmp_calc := gcf.comp_exact_value_correctness_of_stream_eq_some_aux_comp
+        pA ppA ifp_succ_n_fr_ne_zero,
+      have tmp_calc' := gcf.comp_exact_value_correctness_of_stream_eq_some_aux_comp
+        pB ppB ifp_succ_n_fr_ne_zero,
+      rw inv_eq_one_div at tmp_calc tmp_calc',
+      have : fract (1 / ifp_n.fr) ≠ 0, by simpa using ifp_succ_n_fr_ne_zero,
+      -- now unfold the recurrence one step and simplify both sides to arrive at the conclusion
+      field_simp [conts, gcf.comp_exact_value,
+        (gcf.continuants_aux_recurrence s_nth_eq ppred_conts_eq pred_conts_eq), next_continuants,
+        next_numerator, next_denominator, this, tmp_calc, tmp_calc'],
+      ac_refl } }
+end
+
+/--
+Shows the correctness of `comp_exact_value` in case the `int_fract_pair.stream` of the corresponding
+to the continued fraction terminated.
+-/
+lemma comp_exact_value_correctness_of_stream_eq_none
+  (nth_stream_eq_none : int_fract_pair.stream v n = none) :
+  v = (gcf.of v).convergents (n - 1) :=
+begin
+  induction n with n IH,
+  case nat.zero { contradiction }, -- int_fract_pair.stream v 0 ≠ none
+  case nat.succ
+  { rename nth_stream_eq_none succ_nth_stream_eq_none,
+    let g := gcf.of v,
+    change v = g.convergents n,
+    have : int_fract_pair.stream v n = none
+      ∨ ∃ ifp, int_fract_pair.stream v n = some ifp ∧ ifp.fr = 0, from
+      int_fract_pair.succ_nth_stream_eq_none_iff.elim_left succ_nth_stream_eq_none,
+    rcases this with ⟨nth_stream_eq_none⟩ | ⟨ifp_n, nth_stream_eq, nth_stream_fr_eq_zero⟩,
+    { cases n with n',
+      { contradiction }, -- int_fract_pair.stream v 0 ≠ none
+      { have : g.terminated_at n', from
+          gcf.of_terminated_at_n_iff_succ_nth_int_fract_pair_stream_eq_none.elim_right
+          nth_stream_eq_none,
+        have : g.convergents (n' + 1) = g.convergents n', from
+          gcf.convergents_stable_of_terminated n'.le_succ this,
+        rw this,
+        exact (IH nth_stream_eq_none) } },
+    { simpa [nth_stream_fr_eq_zero, gcf.comp_exact_value] using
+      (comp_exact_value_correctness_of_stream_eq_some nth_stream_eq) } }
+end
+
+/-- If `generalized_continued_fraction.of v` terminated at step `n`, then the `n`th convergent is
+exactly `v`. -/
+theorem of_correctness_of_terminated_at (terminated_at_n : (gcf.of v).terminated_at n) :
+  v = (gcf.of v).convergents n :=
+have int_fract_pair.stream v (n + 1) = none, from
+  gcf.of_terminated_at_n_iff_succ_nth_int_fract_pair_stream_eq_none.elim_left terminated_at_n,
+comp_exact_value_correctness_of_stream_eq_none this
+
+/-- If `generalized_continued_fraction.of v` terminates, then its convergent will eventually be
+exactly `v`. -/
+lemma of_correctness_of_terminates (terminates : (gcf.of v).terminates) :
+  ∃ (n : ℕ), v = (gcf.of v).convergents n :=
```

Comment:
You can even say that, for all large enough `n`, you have `v = (gcf.of v).convergents n`, with the syntax
```
∀ᶠ n in at_top, v = (gcf.of v).convergents n
```
(you need to put `open_locale filter` and `open filter` and to `import order.filter.basic` at the beginning of the file. (If you don't know what to do with this, you can `rw eventually_at_top` to get back to something concrete).

Reply:
Uh, nice! I have no clue about filters and so on. Thanks for the hint. Feel free to drop more of these when you discover some in future PRs. :)

I did not need the `open_locale` command though - or is it something I cannot see that is not great the way it is now?